### PR TITLE
1771 raster add file from user zone

### PR DESCRIPTION
**Pull request by hyi**
@pkdash This PR fixed the remaining issue stated in issue #1771 when adding files from federated user zone to an empty resource created in hydroshare zone. I also added unit tests to cover this scenario and made sure tests all passed in my local development environment. Please code review this PR.

### CODE REVIEW

File: hs_geo_raster_resource/tests/test_raster_metadata_user_zone.py [hunk 1]
```diff
@@ -136,3 +136,31 @@ def test_metadata_in_user_zone(self):
 
         # resource core metadata is deleted after resource deletion
         self.assertEqual(CoreMetaData.objects.all().count(), 0)
+
+        # test adding file from user zone to existing empty resource in hydroshare zone
+        # even there is no file uploaded to resource initially, there are default extended
+        # automatically metadata created
+        _, _, metadata, _ = utils.resource_pre_create_actions(
```
Comment (pkdash) [inline hs_geo_raster_resource/tests/test_raster_metadata_user_zone.py]: I don't think this function call is anymore needed as the raster resource type doesn't handle pre resource create signal.

**Comment by pkdash** [comment]
@hyi Does this addresses the issue of the extra tif file?

**Comment by hyi** [comment]
@pkdash Yes, it fixed the issue of the extra tif file, which was caused by the delete function not properly handing deletion of files being added from user zone into a resource stored in hydroshare zone. Since the original tif file was not deleted properly, an extra tif file being added with a random hash code by Django Storage class so as not to override the existing tif file. This PR fixed the delete function to handle this case properly, which fixed the extra tif file issue.

**Comment by hyi** [comment]
@pkdash Have addressed your code review comment and made sure irods federation zone test has passed in my local development environment. See if you can +1 now.

**Comment by pkdash** [comment]
+1. Merge after Jenkins checks are successful.

**Pull request merged**